frontend/src/components/activity/content/Checklist.vue

```diff
@@ -0,0 +1,260 @@
+            </div>
+          </button>
+        </template>
+        <div class="ma-n4">
+          <v-expansion-panels multiple flat accordion>
+            <v-expansion-panel
+              v-for="{ checklist, items } in allChecklists"
+              :key="checklist._meta.self"
+            >
+              <v-expansion-panel-header>
+                <h3>
+                  {{ checklist.name }}
+                  <small class="font-weight-regular">
+                    ({{
+                      selectionContentNode.filter(
+                        (item) =>
+                          checkedItems.includes(item.id) &&
+                          item.checklist()._meta.self === checklist?._meta?.self
+                      ).length
+                    }}
+                    selected)
+                  </small>
+                </h3>
+              </v-expansion-panel-header>
+              <v-expansion-panel-content>
+                <ol>
+                  <ChecklistItem
+                    v-for="{ item } in items.filter(({ item }) => item.parent == null)"
+                    :key="item._meta.self"
+                    :checklist="checklist"
+                    :item="item"
+                    @remove-item="removeItem"
+                    @add-item="addItem"
+                  />
+                </ol>
+              </v-expansion-panel-content>
+            </v-expansion-panel>
+          </v-expansion-panels>
+        </div>
+      </DetailPane>
+    </template>
+  </ContentNodeCard>
+</template>
+
+<script>
+import ContentNodeCard from '@/components/activity/content/layout/ContentNodeCard.vue'
+import { contentNodeMixin } from '@/mixins/contentNodeMixin.js'
+import DetailPane from '@/components/generic/DetailPane.vue'
+import ChecklistItem from './checklist/ChecklistItem.vue'
+import { serverErrorToString } from '@/helpers/serverError.js'
+import { debounce, isEqual, sortBy, uniq } from 'lodash'
+import { computed } from 'vue'
+
+export default {
+  name: 'Checklist',
+  components: { DetailPane, ContentNodeCard, ChecklistItem },
+  mixins: [contentNodeMixin],
+  provide() {
+    return {
+      checkedItems: computed(() => this.checkedItems),
+    }
+  },
+  data() {
+    return {
+      savingRequestCount: 0,
+      dirty: false,
+      showDialog: false,
+      checkedItems: [],
+      uncheckedItems: [],
+      errorMessages: null,
+      debouncedSave: () => null,
+      itemsLoaded: false,
+    }
+  },
+  computed: {
+    campChecklistItems() {
+      return this.api.get().checklistItems().items
+    },
+    selectionContentNode() {
+      return this.campChecklistItems.filter((item) =>
+        this.contentNode
+          .checklistItems()
+          .items.some(({ _meta }) => _meta.self === item._meta.self)
+      )
+    },
+    serverSelection() {
+      return this.selectionContentNode.map((item) => item.id)
+    },
+    allChecklists() {
+      return this.camp.checklists().items.map((checklist) => ({
```

Comment:
should there be some null checks / coalesce operators here?
I fear some NPE's here

Reply:
could be ok, if .item is always available